### PR TITLE
Added options auto_tidyup_dims to disable deletion of 1 dim Systems after multiplication

### PR DESCRIPTION
**Pull request by tacruc**
In the multiplication is a mechanism deleting 1 dimensional Systems. This can lead to problems on Systems having multiple one dimensional Subsystems. Examples are in #823.

Because I coundn't figure out in which cases this deleting mechanism needed, I didn't want to remove it, but provide an option to disable it.

Signed-off-by: Arne Hamann <kontakt+github@arne.email>

### CODE REVIEW

**Comment by ajgpitch** [comment]
This looks fine to me
@nonhermitian do you see any reason not to merge this?

File: qutip/qobj.py [hunk 1]
```diff
@@ -487,7 +487,7 @@ def __mul__(self, other):
                 dims = [self.dims[0], other.dims[1]]
                 out.dims = dims
                 if settings.auto_tidyup: out.tidyup()
-                if (not isinstance(dims[0][0], list) and
+                if (settings.auto_tidyup_dims and not isinstance(dims[0][0], list) and
```
Comment (ajgpitch) [inline qutip/qobj.py]: this line is too long - pep8 wise

**Comment by ajgpitch** [comment]
Actually it needs one pep8 fix please

**Comment by ajgpitch** [comment]
@tacruc for future reference you should really create a new branch on your fork and submit that as a PR for qutip. It's okay in this case, as it is a simple change, but would cause issue if you were possibly making multiple PRs at the same time.

**Comment by ajgpitch** [comment]
If there are no comments on this the I will merge when tests pass

**Comment by tacruc** [comment]
@ajgpitch sorry for the long time without reply. Thanks for the commit with pep8 and the tip with the branch.

**Comment by ajgpitch** [comment]
No worries.
Merged now.
Thanks for the contribution

**Pull request merged**